### DISCUSSION TITLE
Kong plugin store global state

### DISCUSSION THREAD

**Discussion by shettyh** [Kong/kong#8676] [Help] [answered]
Hi,

Have a requirement to store global singleton values in plugin that will only updated when the plugin config changes. Any way to achieve this ?

Want to avoid creating the `resty.ipmatcher` object for each request, want to store this object as a global variable and only update if the plugin config changes

Thanks

**Comment by mayocream** [comment, marked as answer]
An easy way to achieve that is using the Lua week table with GC strategy. 
Example: 

1. Lua cache table
https://github.com/Kong/kong/blob/41060c37cf4ee19926d6201e1a9e12c1915a1d97/kong/plugins/acl/handler.lua#L17-L18

2. Retrieve cache data
https://github.com/Kong/kong/blob/41060c37cf4ee19926d6201e1a9e12c1915a1d97/kong/plugins/acl/handler.lua#L48-L60

**Reply by shettyh** [reply]
Yes tried this, works. Thanks
